extensions/cronet/src/main/java/com/google/android/exoplayer2/ext/cronet/CronetDataSource.java

```diff
@@ -443,16 +443,42 @@ public long open(DataSpec dataSpec) throws HttpDataSourceException {
       throw new OpenException(new InterruptedIOException(e), dataSpec, Status.INVALID);
     }
 
+    // Calculate the content length.
+    if (!isCompressed(responseInfo)) {
```

Comment:
This block was moved up here so that we could know the length of the response. Besides being moved, the block is unchanged.